doc/source/serve/architecture.md

```diff
@@ -24,62 +24,89 @@ There are three kinds of actors that are created to make up a Serve instance:
   the control plane. The Controller is responsible for creating, updating, and
   destroying other actors. Serve API calls like creating or getting a deployment
   make remote calls to the Controller.
-- Router: There is one router per node. Each router is a [Uvicorn](https://www.uvicorn.org/) HTTP
+- HTTP Proxy: There is one HTTP proxy actor on the head node (the location and number of proxies is configurable via the `location` field of `http_options`: {ref}`core-apis`). This actor runs a [Uvicorn](https://www.uvicorn.org/) HTTP
   server that accepts incoming requests, forwards them to replicas, and
   responds once they are completed.
-- Worker Replica: Worker replicas actually execute the code in response to a
+- Worker Replicas: Actors that actually execute the code in response to a
   request. For example, they may contain an instantiation of an ML model. Each
-  replica processes individual requests from the routers (they may be batched
+  replica processes individual requests from the HTTP proxy (these may be batched
   by the replica using `@serve.batch`, see the [batching](serve-batching) docs).
 
 ## Lifetime of a Request
 
-When an HTTP request is sent to the router, the follow things happen:
+When an HTTP request is sent to the HTTP proxy, the following things happen:
 
-- The HTTP request is received and parsed.
-- The correct deployment associated with the HTTP url path is looked up. The
+1. The HTTP request is received and parsed.
+2. The correct deployment associated with the HTTP URL path is looked up. The
   request is placed on a queue.
-- For each request in a deployment queue, an available replica is looked up
-  and the request is sent to it. If there are no available replicas (there
-  are more than `max_concurrent_queries` requests outstanding), the request
-  is left in the queue until an outstanding request is finished.
+3. For each request in a deployment's queue, an available replica is looked up in round-robin fashion
+  and the request is sent to it. If there are no available replicas (i.e. there
+  are more than `max_concurrent_queries` requests outstanding at each replica), the request
+  is left in the queue until a replica becomes available.
 
-Each replica maintains a queue of requests and executes one at a time, possibly
-using asyncio to process them concurrently. If the handler (the function for the
-deployment or `__call__`) is `async`, the replica will not wait for the
-handler to run; otherwise, the replica will block until the handler returns.
+Each replica maintains a queue of requests and executes requests one at a time, possibly
+using `asyncio` to process them concurrently. If the handler (the deployment function or the `__call__` method of the deployment class) is declared with `async def`, the replica will not wait for the
+handler to run.  Otherwise, the replica will block until the handler returns.
```

Comment:
Not "the replica will not wait for the handler to run." but "the replica will run the asyncio methods concurrently with other invocations in the same event loop.